parser-typechecker/src/U/Codebase/Branch/Diff.hs

```diff
@@ -83,56 +88,71 @@ data NameChanges = NameChanges
   }
 
 instance Semigroup NameChanges where
-  (NameChanges a b c d) <> (NameChanges a2 b2 c2 d2) =
+  NameChanges a b c d <> NameChanges a2 b2 c2 d2 =
     NameChanges (a <> a2) (b <> b2) (c <> c2) (d <> d2)
 
 instance Monoid NameChanges where
   mempty = NameChanges mempty mempty mempty mempty
 
+-- | A name-based diff for namespaces `N1` and `N2` is (for both terms and types) a relation between references, where
+-- `a R b` if:
+--
+--   1. `a` has name `n` in `N1`, and `b` has the same name `n` in `N2`
+--   2. `a` != `b`
+data NameBasedDiff = NameBasedDiff
+  { terms :: Relation Reference Reference,
+    types :: Relation Reference Reference
+  }
+  deriving stock (Generic, Show)
+
+instance Monoid NameBasedDiff where
+  mempty = NameBasedDiff mempty mempty
+  mappend = (<>)
```

Comment:
I don't think we need `mappend` defined anymore, in fact I think some GHC's complain about it if you do?

Reply:
Oh, whoops!